cc/tflite_micro/README.md

```diff
@@ -0,0 +1,196 @@
+|   |   |   hello_world tflite model app.
+|   |   |
+|   |   |-- BUILD
+|   |   |   Provide common dependencies shared by all tflite model apps.
+|   |   |
+|   |   |-- start.S
+|   |   |   Used by tflite model apps to generate freestanding and executable binary
+|   |   |   to run on Linux.
+|   |
+|   |-- include
+|   |   Provide Oak-specific declarations to tflite model apps.
+|   |
+|   |-- BUILD
+|   |   Provide Oak-specific build options and shared headers/sources/deps
+|   |   i.e. Oak DebugLog and TFLM-on-Oak library.
+|   |
+|   |-- build_defs.bzl
+|   |   Provide Oak-specific compiler options, linker options and tool macros.
+|   |
+|   |-- debug_log.cc
+|   |   Implement Oak-specific debug logging required by tflm micro-printf.
+|   |
+|   |-- tools
+|   |   |-- BUILD
+|   |   |   Tool dep to convert tflite model to cc arrays in bazel build process.
+|   |   |
+|   |   |-- generate_cc_arrays.py
+|   |   |   Convert a tflite model to a static C array and generate corresponding header and
+|   |   |   source files which can be specified in cc_binary/cc_library rules. Sourced and
+|   |   |   modified from tflm's generate_cc_arrays.py.
+|   |   |
+|   |   |-- update_tflm.sh
+|   |   |   Tool for upgrading/downgrading tflm sources and dependencies.
+|
+|-- libc
+|   A pico libc linked by tflm and only required apis are implemented.
+|   bsearch and optimized (by SIMD instruction set i.e. SSE2/SSSE3/SSE4/AVX2) string apis
+|   are sourced and modified from Android bionic libc*.
+|   misc.cc implements apis required by tflm and releated/specific to Oak trusted execution environment.
+|   Other sources are sourced from Google nanolibc**, which currently misses bsearch,
+|   string-to-{float|double} and most apis are implemented in a highly non-optimized
+|   way for better portability.
+|
+|-- libgcc
+|   A pico libgcc implements clrsb GCC builtin function required by tflm.
+|
+|-- libm
+|   A pico libm implements a set of complementary math apis required by tflm.
+|
+|   For freestanding model app binaries running on Linux, compiler links static libm by default (-lm)
+|   to pull in required math apis in addition to the math apis supplemented by this libm.
+|
+|   For freestanding model app binaries running in Oak server, rustc compiler would pull in required
+|   math apis from Oak libm.rs in addition to the math apis supplemented by this libm.
+|
+```
+
+\*
+[Android bionic libc](https://android.googlesource.com/platform/bionic/+/refs/heads/master)\
+\*\* [Google nanolibc](https://github.com/google/nanolibc)
+
+## Upgrade/Downgrade Tensorflow Lite for Microcontrollers
+
+We generate a clean set of tflm source tree via its built-in tool
+[create_tflm_tree.py](https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/tools/project_generation/create_tflm_tree.py)
+that cuts ~50% sources to be compiled, which also implies less external
+dependency errors to fix in a bare-metal and freestanding binary development
```

Comment:
```suggestion
dependency errors to fix in a freestanding binary development
```

we are trying not to use bare-metal, as it is not technically accurate, and it can be confused with what some people mean as "running on the actual machine instead of a VM".

Reply:
Good point! Removed it.